### PR TITLE
Fix the `version` and `source` references for examples in documentation.

### PR DESCRIPTION
**Pull request by NotGordias**
- fixes the plugin source (from github.com/hashicorp/linode to github.com/hashicorp/packer-plugin-linode) in example
- bumped the version in the example since `1.0.1` is the latest release

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -19,8 +19,8 @@ Then, run [`packer init`](https://www.packer.io/docs/commands/init).
 packer {
   required_plugins {
     linode = {
-      version = ">= 1.0.0"
-      source  = "github.com/hashicorp/linode"
+      version = ">= 1.0.1"
+      source  = "github.com/hashicorp/packer-plugin-linode"
```
Comment (nywilken) [inline docs/README.md:23]: The source reference is not the actual full URL. The common prefix `packer-plugin-` is automatically applied during the init process.
```suggestion
      source  = "github.com/hashicorp/linode"
```


 By including it in the source reference you are actually telling Packer to try and download the plugin named `packer-plugin-packer-plugin-linode`, which will result in the following error. 

```
Plugin source "github.com/hashicorp/linode" has a type with the prefix
"packer-plugin-", which isn't valid. Although that prefix is often used in the
names of version control repositories for Packer plugins, plugin source strings
should not include it.

Did you mean "github.com/hashicorp/linode"?
```

File: docs/builders/linode.mdx [hunk 2]
```diff
@@ -30,8 +30,8 @@ Then, run [`packer init`](https://www.packer.io/docs/commands/init).
 packer {
   required_plugins {
     linode = {
-      version = ">= 0.0.1"
-      source  = "github.com/hashicorp/linode"
+      version = ">= 1.0.1"
+      source  = "github.com/hashicorp/packer-plugin-linode"
```
Comment (nywilken) [inline docs/builders/linode.mdx:34]:
```suggestion
      source  = "github.com/hashicorp/linode"
```

File: example/basic_linode.pkr.hcl [hunk 3]
```diff
@@ -1,8 +1,8 @@
 packer {
   required_plugins {
     linode = {
-      version = ">= 1.0.0"
-      source  = "github.com/hashicorp/linode"
+      version = ">= 1.0.1"
+      source  = "github.com/hashicorp/packer-plugin-linode"
```
Comment (nywilken) [inline example/basic_linode.pkr.hcl:5]:
```suggestion
      source  = "github.com/hashicorp/linode"
```

**Review by nywilken — CHANGES_REQUESTED** [review]
Thanks for updating the documentation to use the latest version. There is a small issue with the updated source reference that needs to be addressed before we can merge. 

I left a detail comment around the suggested change. Please let us know if you have any questions.

File: docs/README.md (hunk 1 above)
Comment (NotGordias) [reply docs/README.md]: Ah, good to know. Thank you for the detailed explanation!

**Comment by NotGordias** [comment]
Changes applied, thank you!

**Review by nywilken — APPROVED** [review]
LGTM

**Pull request merged**